### PR TITLE
[TRAFODION-1656]Remove DISCONNECTED& refactor some code

### PR DESCRIPTION
**Pull request by kevinxu021**
It's useless to be DISCONNECTED since nothing is done between DISCONNECTED and AVALIBLE.
And seems to be not reasonable to sleep 30s.
Tested on while(true){connect/close} without sleeping and it successfully did hundreds times.

### CODE REVIEW

**Comment by DaveBirdsall** [comment]
Looks OK to me.

File: dcs/src/main/java/org/trafodion/dcs/master/listener/ConnectReply.java [hunk 1]
```diff
@@ -163,29 +163,21 @@ boolean buildConnectReply  (Header hdr, ConnectionContext cc, SocketAddress clie
...
-                    else {
-                        found = true;
-                        break;
-                    }
-                }
-                else
-                    continue;
+                if(LOG.isDebugEnabled())
+                    LOG.debug(clientSocketAddress + ": " + "server selected in search 1 " + server );
+                data = isServerAvalible(nodeRegisteredPath);
```
Comment (arvind-narain) [inline dcs/src/main/java/org/trafodion/dcs/master/listener/ConnectReply.java]: Thanks for the refactor Kevin. Code looks good now.
Question - Doesn't seem to be a difference between the new log message (175) for this case and the next case (191) - intentional ? I think earlier the index was being printed.

Code is good to be checked in. The next set of change (for disconnect) will allow tests to pass without holding onto the connections - thanks for that.

**Comment by DaveBirdsall** [comment]
Is this ready to commit now?

**Comment by kevinxu021** [comment]
Arvind, any more comments? If no, please merge it.

**Comment by arvind-narain** [comment]
Thanks Kevin for the change - will merge it now.

**Pull request merged**